src/main/java/org/apache/aurora/scheduler/configuration/ConfigurationManager.java

```diff
@@ -170,12 +164,24 @@ public IJobConfiguration validateAndPopulate(IJobConfiguration job)
       throw new TaskDescriptionException("Job configuration must have taskConfig set.");
     }
 
+    JobConfiguration builder = job.newBuilder();
+    
     if (job.getInstanceCount() <= 0) {
       throw new TaskDescriptionException("Instance count must be positive.");
+    } else {
```

Comment:
Minor: Can we make instanceCount optional, in which case we require instances to be present?